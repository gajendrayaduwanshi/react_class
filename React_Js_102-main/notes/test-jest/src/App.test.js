import React from 'react';
import renderer from 'react-test-renderer';
import Link from './App';
import TestRenderer from 'react-test-renderer';
    
const {act} = TestRenderer;

test('Link changes on hovering', () =>{
    const component = renderer.create(
         <Link page="http://www.google.com">Google</Link>,
    );    

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => tree.props.onMouseEnter());

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => tree.props.onMouseLeave());    

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
