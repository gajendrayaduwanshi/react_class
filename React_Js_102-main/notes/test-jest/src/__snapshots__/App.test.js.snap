// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes on hovering 1`] = `
<a
  className="narmal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes on hovering 2`] = `
<a
  className="hovered"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Link changes on hovering 3`] = `
<a
  className="narmal"
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
